src/main/java/org/jfrog/hudson/pipeline/common/executors/CollectEnvExecutor.java

```diff
@@ -0,0 +1,46 @@
+package org.jfrog.hudson.pipeline.common.executors;
+
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.remoting.VirtualChannel;
+import jenkins.MasterToSlaveFileCallable;
+import org.jfrog.hudson.pipeline.common.types.buildInfo.Env;
+
+import java.io.File;
+import java.io.IOException;
+
+public class CollectEnvExecutor implements Executor {
+    private transient TaskListener listener;
+    private transient FilePath ws;
+    private Run build;
+    private Env env;
+    private EnvVars envVars;
```

Comment:
```suggestion
    private final transient TaskListener listener;
    private final transient FilePath ws;
    private final Run<?, ?> build;
    private final Env env;
    private final EnvVars envVars;
```

Reply:
'final' may break with the 'transient' keyword.